drivers/video/video_arducam_mega.c

```diff
@@ -0,0 +1,1405 @@
+		{VIDEO_COLORFX_GRASS_GREEN, MEGA_COLOR_FX_GRASS_GREEN},
+		{VIDEO_COLORFX_VIVID, MEGA_COLOR_FX_OVER_EXPOSURE},
+	};
+
+	uint8_t mega_effect = MEGA_COLOR_FX_NONE;
+	bool supported = false;
+
+	for (size_t i = 0; i < ARRAY_SIZE(fx_map); ++i) {
+		if (fx_map[i].video_fx == effect) {
+			mega_effect = fx_map[i].mega_fx;
+			supported = true;
+			break;
+		}
+	}
+
+	if (!supported) {
+		LOG_ERR("Unsupported color effect: %d", effect);
+		return -ENOTSUP;
+	}
+
+	return arducam_mega_write_reg_wait(&cfg->bus, CAM_REG_COLOR_EFFECT_CONTROL, mega_effect, 3);
+}
+
+static int arducam_mega_set_output_format(const struct device *dev, int output_format)
+{
+	const struct arducam_mega_config *cfg = dev->config;
+	uint8_t format_val;
+
+	switch (output_format) {
+	case VIDEO_PIX_FMT_JPEG:
+		format_val = MEGA_PIXELFORMAT_JPG;
+		break;
+	case VIDEO_PIX_FMT_RGB565:
+		format_val = MEGA_PIXELFORMAT_RGB565;
+		break;
+	case VIDEO_PIX_FMT_YUYV:
+		format_val = MEGA_PIXELFORMAT_YUV;
+		break;
+	default:
+		LOG_ERR("Image format not supported");
+		return -ENOTSUP;
+	}
+
+	int ret = arducam_mega_write_reg_wait(&cfg->bus, CAM_REG_FORMAT, format_val, 3);
+
+	if (ret < 0) {
+		return ret;
+	}
+
+	ret = arducam_mega_await_bus_idle(&cfg->bus, 30);
+	if (ret < 0) {
+		LOG_ERR("Bus idle wait failed after setting output format");
+	}
+
+	return ret;
+}
+
+static int arducam_mega_set_JPEG_quality(const struct device *dev, enum mega_image_quality qc)
+{
+	const struct arducam_mega_config *cfg = dev->config;
+	struct arducam_mega_data *drv_data = dev->data;
+
+	LOG_DBG("%s: %d", __func__, qc);
+
+	if (drv_data->fmt.pixelformat != VIDEO_PIX_FMT_JPEG) {
+		LOG_ERR("Image format does not support setting JPEG quality");
+		return -ENOTSUP;
+	}
+
+	return arducam_mega_write_reg_wait(&cfg->bus, CAM_REG_IMAGE_QUALITY, qc, 3);
+}
+
+static int arducam_mega_set_white_bal_enable(const struct device *dev, int enable)
+{
+	const struct arducam_mega_config *cfg = dev->config;
+
+	uint8_t reg = CTR_WHITEBALANCE;
+
+	if (enable) {
+		reg |= 0x80;
+	}
+
+	int ret = arducam_mega_write_reg_wait(&cfg->bus, CAM_REG_EXPOSURE_GAIN_WHITEBAL_ENABLE, reg,
+					      3);
+
+	if (ret < 0) {
+		return ret;
+	}
+
+	ret = arducam_mega_await_bus_idle(&cfg->bus, 10);
+	if (ret < 0) {
+		LOG_ERR("Bus idle wait failed after setting white balance");
+	}
+
+	return ret;
+}
+
+static int arducam_mega_set_white_bal(const struct device *dev, enum mega_white_balance level)
+{
+	const struct arducam_mega_config *cfg = dev->config;
+
+	return arducam_mega_write_reg_wait(&cfg->bus, CAM_REG_WHITEBALANCE_CONTROL, level, 3);
+}
+
+static int arducam_mega_set_gain_enable(const struct device *dev, int enable)
+{
+	const struct arducam_mega_config *cfg = dev->config;
+
+	uint8_t reg = CTR_GAIN;
+
+	if (enable) {
+		reg |= 0x80;
+	}
+
+	int ret = arducam_mega_write_reg_wait(&cfg->bus, CAM_REG_EXPOSURE_GAIN_WHITEBAL_ENABLE, reg,
+					      3);
+
+	if (ret < 0) {
+		return ret;
+	}
```

Comment:
Would it be possible to group the error handling with the function call?

This might imply `int ret;` declared at the top.

```suggestion
	ret = arducam_mega_write_reg_wait(&cfg->bus, CAM_REG_EXPOSURE_GAIN_WHITEBAL_ENABLE, reg, 3);
	if (ret < 0) {
		return ret;
	}
```